File: Cite.EvalIt.WebApp/src/app/core/query/rank-recalculation-task.lookup.ts
import { IsActive } from '@app/core/enum/is-active.enum';
import { RankRecalculationTaskStatus } from '@app/core/enum/rank-recalculation-task-status.enum';
import { Lookup } from '@common/model/lookup';
import { Guid } from '@common/types/guid';

export class RankRecalculationTaskLookup extends Lookup implements RankRecalculationTaskFilter {
	ids: Guid[];
	excludedIds: Guid[];
	requestingUserIds: Guid[];
	taskStatuses: RankRecalculationTaskStatus[];
	isActive: IsActive[];
	createdAfter?: Date;

	constructor() {
		super();
	}
}

export interface RankRecalculationTaskFilter {
	ids: Guid[];
	excludedIds: Guid[];
	requestingUserIds: Guid[];
	taskStatuses: RankRecalculationTaskStatus[];
	isActive: IsActive[];
	createdAfter?: Date;
}

File: Cite.EvalIt/Query/RankRecalculationTaskLookup.cs
using Cite.EvalIt.Common;
using Cite.Tools.Data.Query;
using System;
using System.Collections.Generic;

namespace Cite.EvalIt.Query
{
	public class RankRecalculationTaskLookup : Lookup
	{
		public List<Guid> Ids { get; set; }
		public List<Guid> ExcludedIds { get; set; }
		public List<Guid> RequestingUserIds { get; set; }
		public List<RankRecalculationTaskStatus> TaskStatuses { get; set; }
		public List<IsActive> IsActive { get; set; }
		public DateTime? CreatedAfter { get; set; }

		public RankRecalculationTaskQuery Enrich(QueryFactory factory)
        {
			RankRecalculationTaskQuery query = factory.Query<RankRecalculationTaskQuery>();

			if (this.Ids != null) query.Ids(this.Ids);
			if (this.ExcludedIds != null) query.ExcludedIds(this.ExcludedIds);
			if (this.RequestingUserIds != null) query.RequestingUserIds(this.RequestingUserIds);
			if (this.TaskStatuses != null) query.Status(this.TaskStatuses);
			if (this.IsActive != null) query.IsActive(this.IsActive);
			if (this.CreatedAfter != null) query.CreatedAfter(this.CreatedAfter);

			this.EnrichCommon(query);

			return query;
		}
	}
}
